SPECS/qemu/qemu.spec

```diff
@@ -1,64 +1,257 @@
-# Disable stripping - strip is trying to strip binaries (.img, .elf, etc.) built for different architectures.
+Vendor:         Microsoft Corporation
+Distribution:   Azure Linux
+# Provide a way to skip tests via rpmbuild `--without`
+# This makes it easier to skip tests in copr repos, where
+# the qemu test suite is historically flakey
+%bcond_without check
+
+# This spec is for AzLinux
+%global azl 3
+%global azl_no_ui 1
+
 %global __strip /bin/true
 %global libfdt_version 1.6.0
 %global libseccomp_version 2.4.0
 %global libusbx_version 1.0.23
-%global meson_version 0.58.2
+%global meson_version 0.61.3
 %global usbredir_version 0.7.1
-%global ipxe_version 1.20.1
+%if 0%{?azl}
+%global ipxe_version 1.21.1
+%else
+%global ipxe_version 20200823-5.git4bd064de
+%endif
+%if 0%{?azl}
 %global excluded_targets moxie-softmmu
+%endif
 %global have_memlock_limits 0
 %global need_qemu_kvm 0
+%ifarch %{ix86}
+%global kvm_package   system-x86
+# need_qemu_kvm should only ever be used by x86
+%global need_qemu_kvm 1
+%endif
 %ifarch x86_64
 %global kvm_package   system-x86
 # need_qemu_kvm should only ever be used by x86
 %global need_qemu_kvm 1
 %endif
+%ifarch %{power64}
+%global have_memlock_limits 1
+%global kvm_package   system-ppc
+%endif
+%ifarch s390x
+%global kvm_package   system-s390x
+%endif
+%ifarch armv7hl
+%global kvm_package   system-arm
+%endif
 %ifarch aarch64
 %global kvm_package   system-aarch64
 %endif
+%ifarch %{mips}
+%global kvm_package   system-mips
+%endif
+%ifarch riscv64
+%global kvm_package   system-riscv
+%endif
+
 %global modprobe_kvm_conf %{_sourcedir}/kvm.conf
-%ifarch x86_64
+%ifarch s390x
+    %global modprobe_kvm_conf %{_sourcedir}/kvm-s390x.conf
+%endif
+%ifarch %{ix86} x86_64
     %global modprobe_kvm_conf %{_sourcedir}/kvm-x86.conf
 %endif
+
 %global tools_only 0
+
+
+%global user_static 1
```

Comment:
Random comment -- I know it's not turning this feature on, but I'm glad to see having at least latent support in the spec for user_static. This is one of the features required for Azure Linux to support running cross-architecture Docker containers (e.g., running an x86_64 container on aarch64, or vice versa).